### ISSUE TITLE
Custom tax driver missing some key data

### ISSUE THREAD

**Issue by ryanmitchell** [getcandy/getcandy#173]
- GetCandy version: 2.0-beta9
- Laravel Version: 8.83.0
- PHP Version: 8.0.13
- Database Driver & Version: Mysql 8

### Expected Behaviour:

I have a basic custom tax driver in place to test the functionality, however there are some shortcomings and issues.

1. I need to be able to access the cart line. Providers such as tax jar / AvaTax return individual lines so you need to map their response to return the right breakdown for the line being calculated. As it stands I can do a match on the line using the purchasable sku, but its possible to have the same sku in the cart on different lines as the meta is different, so this isnt scaleable. Really I need to pass a line id to them and map it to the cart line in the tax driver getBreakdown() method.

2. Its unclear what content getBreakdown should return from the docs ... I understand its a collection, but of what? From looking around the code it seems to be:

`[ 'name' => '', 'total' => '', 'percentage' => '', 'tax_rate_id' => '']`

But then I have to make it a collection of collections for it to work, and even it still errors out in `Managers/CartManager` line 119 with the error `Attempt to read property "taxRate" on string`. All guidance gratefully received :)

### Actual Behaviour:

As above.

### Steps To Reproduce:

As above.

**Comment by ryanmitchell** [comment]
Yep adding setCartLine() sounds like a good approach. It may be worth getting people to extend a base class rather than an interface so that adding things doesnt break implementations (not such an issue now but would be later).

On the breakdown I think its easier for you to explicitly say what you expect a tax driver to return from the getBreakdown method.

**Comment by glennjacobs** [comment]
Yeah, I agree, we need to be specific if we're currently not. @alecritson maybe we need a DTO or similar?

**Comment by alecritson** [comment]
@glennjacobs I think that probably makes sense. I can start work on a PR as it should be a minor thing.
